src/shared_gui_components/OSGridView.cpp

```diff
@@ -364,6 +370,34 @@ void OSGridView::recreateAll() {
   }
 }
 
+int OSGridView::layoutIndex(int row) const {
+  return row / NUM_ROWS_PER_GRIDLAYOUT;
+}
+
+int OSGridView::rowInLayout(int row) const {
+  return row % NUM_ROWS_PER_GRIDLAYOUT;
+}
```

Comment:
```suggestion
constexpr int OSGridView::layoutIndex(int row) const {
  return row / NUM_ROWS_PER_GRIDLAYOUT;
}

constexpr int OSGridView::rowInLayout(int row) const {
  return row % NUM_ROWS_PER_GRIDLAYOUT;
}
```

Reply:
I don't think this can be constexpr because row is not known at compile time

Reply:
The constexpr specifier declares that it is possible to evaluate the value of the function or variable at compile time. Which it can.

```c++
#include <iostream>

constexpr int NUM_ROWS_PER_GRIDLAYOUT = 51;

constexpr int layoutIndex(int row) {
  return row / NUM_ROWS_PER_GRIDLAYOUT;
}

int main() {
  int number;
  std::cout << "Please enter a row: ";
  std::cin >> number;

  int idx = layoutIndex(number);
  std::cout << "layoutIndex=" << idx << '\n';
  return idx;
}
```

```
$ g++ -std=c++17 main.cpp -o test && ./test
Please enter a row: 109
layoutIndex=2
```

Reply:
Anyways, that's just pedantism on my part :)

Reply:
![image](https://user-images.githubusercontent.com/1327850/124337779-88e5bc00-db61-11eb-919a-c2151987fc08.png)